robot_sort/robot_sort.py

```diff
@@ -81,32 +84,68 @@ def set_light_on(self):
         Turn on the robot's light
         """
         self._light = "ON"
+
     def set_light_off(self):
         """
         Turn off the robot's light
         """
         self._light = "OFF"
+
     def light_is_on(self):
         """
         Returns True if the robot's light is on and False otherwise.
         """
         return self._light == "ON"
 
+    def back_to_beginning(self):
+        if self.can_move_left():
+            self.move_left()
+        else:
+            self.set_light_off()
+            return
+        return self.back_to_beginning()
+
     def sort(self):
         """
         Sort the robot's list.
         """
-        # Fill this out
-        pass
+        if self.can_move_right() == False:
```

Comment:
Excellent work getting the tests working, however, creating a new method breaks the rules. No worries though, here's a solution. Try to rework yours to not use the back_to_beginning method

  """
        Sort the robot's list.
        """
        # Selection Sort solution
        self.swap_item()
        self.set_light_on()
        
        while self.light_is_on():
            # move right and find the smallest item that has not been sorted
            while self.move_right():
                if self.compare_item() > 0:
                    self.swap_item()
            # if the robot is at the end of the list and all items are sorted, we're done
            if self.compare_item() is None and self.can_move_right() is False:
                # place the current item down in the list and turn light off to indicate the list is sorted
                self.swap_item()
                self.set_light_off()
                break
            else:
                # find the spot in the list that is holding None
                # place the item in that position
                # move right and swap
                while self.move_left():
                    if self.compare_item() is None:
                        self.swap_item()
                        self.move_right()
                        self.swap_item()
                        break